pkg/sql/opt/optbuilder/fk_cascade.go

```diff
@@ -189,6 +189,9 @@ func tryNewOnDeleteFastCascadeBuilder(
 		if memo.CanBeCompositeSensitive(md, &sel.Filters) {
 			return nil, false
 		}
+		if sel.Relational().HasSubquery {
```

Comment:
I considered leaving a TODO here to explain that a cascade fast path could be planned for non-correlated subqueries if we correctly replaced references to tables in Scans and other relational expressions. But I don't think there's much value in supporting that and it seems very difficult to get correct, so I doubt we'd ever want to support that.